[B5]Migrate Profile Case Search page to b5

jingcheng16 opened the pull request:
## Technical Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
Before:
![image](https://github.com/dimagi/commcare-hq/assets/39149002/fe626074-63eb-4b8b-83d6-cdb379eac24a)
![image](https://github.com/dimagi/commcare-hq/assets/39149002/dfbd2608-8436-4c80-bb25-bea6a4b7e984)
![image](https://github.com/dimagi/commcare-hq/assets/39149002/7f0e1c0d-e200-4ec4-9dd9-bc8fb08dc788)
![image](https://github.com/dimagi/commcare-hq/assets/39149002/ab879163-3ee5-414d-87bf-2fce39cadf47)

After:
![image](https://github.com/dimagi/commcare-hq/assets/39149002/05d7f1c5-d428-4608-9ce1-0c9541679fd7)
![image](https://github.com/dimagi/commcare-hq/assets/39149002/88bbea93-ff8d-46f8-8b3e-3452df452f9a)
![image](https://github.com/dimagi/commcare-hq/assets/39149002/8f37f825-7c26-46ba-891b-64d160f91393)


## Feature Flag
<!-- If this is specific to a feature flag, which one? -->
(required) Turn on the "[Enable extension sync](https://www.commcarehq.org/hq/flags/edit/extension_sync/?__hstc=240960668.2195ac63c0056c39905e6882865ae8e3.1597171287126.1616447050240.1616526330286.228&__hssc=240960668.8.1616526330286&__hsfp=1126783430)" feature flag

(required) Turn on the "[Enable synchronous mobile searching and case claiming](https://www.commcarehq.org/hq/flags/edit/search_claim/)" feature flag
## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
This page is for internal use only.
Tested locally and on staging.
### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->
Yes (bootstrap 5 diffs)

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
No QA

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

jingcheng16 committed:
Migrate ProfileCaseSearchView to b5

jingcheng16 committed:
Update profile case search html to b5 stylings

jingcheng16 committed:
Bootstrap 5 Migration - Marked template 'case_search/profile_case_search.html' as complete and un-split files.

gherceg commented on corehq/apps/case_search/templates/case_search/bootstrap3/profile_case_search.html:
```diff
@@ -1,82 +0,0 @@
-{% extends 'hqwebapp/bootstrap3/base_section.html' %}
```
Is this also the appropriate time to remove `case_search/js/bootstrap3/profile_case_search.js`?

orangejenny replied:
Yes, good point. Since the template is being moved/renamed, also seems like a good time to move `case_search/js/bootstrap5/profile_case_search` to `case_search/js/profile_case_search`.

jingcheng16 replied:
Thanks Graham and Jenny! I just unsplit `profile_case_search.js`

gherceg reviewed (commented):
Looks good but given I don't know what I'm talking about, holding off on approval.

jingcheng16 committed:
rebuild bootstrap 5 diffs

orangejenny commented on corehq/apps/case_search/templates/case_search/profile_case_search.html:
```diff
@@ -62,7 +62,7 @@ <h3>Profile Slow Case Searches</h3>
     </script>
 
-    <div class="panel-group" id="queries-accordion" data-bind="foreach: results().queries">
+    <div class="card-group" id="queries-accordion" data-bind="foreach: results().queries">
```
Is this doing anything? I think `panel-group` is an HQ style, not a bootstrap style, and I don't see `card-group` in B5 docs.

jingcheng16 replied:
Hi Jenny, change `panel-group` to `card-group` seems have no affect. But here is the doc for `card-group`:https://getbootstrap.com/docs/5.0/components/card/#card-groups

orangejenny replied:
Cool, thank you. If it's not doing anything, let's leave it off. That class looks like it's meant for cards that will be displayed side-by-side, which we're not doing here.

jingcheng16 replied:
You're right. Sorry for the oversight. Changed here: becd3135baad122fa5641b250e948e123a81b047

orangejenny reviewed (commented):
Just one question.

jingcheng16 committed:
Bootstrap 5 Migration - Marked js file 'case_search/js/profile_case_search.js' as complete and un-split files.

jingcheng16 committed:
rebuild bootstrap5 diff

jingcheng16 committed:
Change card-group back to panel-group

Thread: https://github.com/dimagi/commcare-hq/pull/34459#discussion_r1571423731

orangejenny reviewed (approved):
beautiful

jingcheng16 committed:
Add margin to the card
